pycbc/events/stat.py

```diff
@@ -1828,6 +1833,12 @@ def __init__(self, sngl_ranking, files=None, ifos=None, **kwargs):
         """
         ExpFitFgBgNormStatistic.__init__(self, sngl_ranking, files=files,
                                          ifos=ifos, **kwargs)
+        self.find_kdes()
+
```

Comment:
I think the `for kname in self.kde_names:` can be moved here rather than being put inside find_kdes, then when reusing from another class one can have the same loop and `ExpFitFgBgKDEStatistic.assign_kdes(kname)` inside the loop

Reply:
That's the exact same change I am just testing!

Reply:
Hmm, it seems not ... I said move the loop over self.kde_names _here_, ie into the init method, not roll it into find_kdes.  If we wanted to remove the assign_kdes function entirely I would have requested that.

Reply:
Ah okay, I have reverted the last change and added in your suggestion.

It is a little more repetition, but seems to be a bit clearer as to what is happening

Reply:
Right, I am thinking we want to be as simple as possible if reusing methods from another class, so avoiding nested methods and keeping each one short is preferable.